src/auth.ts

```diff
@@ -51,12 +53,15 @@ export const authOptions: NextAuthOptions = {
   ],
 
   callbacks: {
-    // If user does not exist already, redirect to registration page.
-    // TBD: Add a URL parameter or an intermediate screen to explain?
     async signIn({ user }) {
       if (user.email) {
         const existingUser = await getUserByEmail(user.email);
-        return existingUser ? true : Route.register;
+        // If user does not exist already, redirect back to sign-in page
+        // with an error explainer and the option to register instead.
+        if (!existingUser) {
+          return `/auth/sign-in?error=UserNotFound&email=${user.email}`;
```

Comment:
Hodil bych `UserNotFound` do konstanty, než to hard-codit napříč kódem. :)

Reply:
Hm, nakonec jsem to stejně vrátil do původního stavu, protože ten import konstanty ze `src/auth.ts` z client-side komponenty dělal nějaký bordel a nechce se mně v tom bádat 💩

Reply:
To je prave ta bariera client/server. Nemuzes do klienta importovat nic ciste serveroveho (z Node), ani prenesene. Proto mam v codebase "neutralni" soubory jako xyztypes.ts, odkud vsude importuju typy a konstanty a tak.

(vim vim, vic souboru)